plugins/processors/starlark/starlark.go

```diff
@@ -40,6 +40,41 @@ type Starlark struct {
 	applyFunc *starlark.Function
 	args      starlark.Tuple
 	results   []telegraf.Metric
+	state     State
+}
+
+// The struct that defines the shared state used
+type State struct {
+	content map[string]interface{}
+}
+
+// Load from the shared state, the value corresponding to the given key
+func (s *State) Load(key string) starlark.Value {
+	var value = s.content[key]
+	if value == nil {
+		return starlark.None
+	}
+	switch v := value.(type) {
```

Comment:
do we need to handle cases for `[]telegraf.Metric`, and any other nested Telegraf metric?

Reply:
@ssoroka I have added all the supported [data types ](https://github.com/google/starlark-go/blob/master/doc/spec.md#data-types)